src/Hive/Controllers/UserController.cs

```diff
@@ -50,7 +63,9 @@ public class UserController : ControllerBase
         private readonly PermissionsManager<PermissionContext> permissions;
 
         private const string RenameActionName = "hive.user.rename";
+        private const string GetUserInfoActionName = "hive.user.info";
         private PermissionActionParseState renameParseState;
+        private PermissionActionParseState getUserParseState;
```

Comment:
Both of these should be `[ThreadStatic]`, both because they are not intended to support multi-threaded access, and because all of the controller types are transitively scoped, and are re-created for each request.